turbopack/crates/turbo-tasks/src/vc/mod.rs

```diff
@@ -34,18 +34,98 @@ use crate::{
     CellId, RawVc, ResolveTypeError, SharedReference, ShrinkToFit,
 };
 
-/// A Value Cell (`Vc` for short) is a reference to a memoized computation
-/// result stored on the heap or in persistent cache, depending on the
-/// Turbo Engine backend implementation.
+/// A "Value Cell" (`Vc` for short) is a reference to a memoized computation result stored on the
+/// heap or in persistent cache, depending on the Turbo Engine backend implementation.
 ///
-/// In order to get a reference to the pointed value, you need to `.await` the
-/// [`Vc<T>`] to get a [`ReadRef<T>`][crate::ReadRef]:
+/// In order to get a reference to the pointed value, you need to `.await` the [`Vc<T>`] to get a
+/// [`ReadRef<T>`][`ReadRef`]:
 ///
 /// ```
 /// let some_vc: Vc<T>;
 /// let some_ref: ReadRef<T> = some_vc.await?;
 /// some_ref.some_method_on_t();
 /// ```
+///
+/// `Vc`s are similar to a [`Future`] or a Promise with a few key differences:
+///
+/// - Unlike futures (but like promises), the work that a `Vc` represents begins execution
+///   immediately, even if the `Vc` is not `await`ed.
```

Comment:
Really? I thought that wasn't the case to lazily execute only needed values?

Reply:
Follow up discussion: https://github.com/vercel/next.js/pull/72847#discussion_r1844408367

![Screenshot 2024-11-25 at 3.55.22 PM.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/HAZVitxRNnZz8QMiPn4a/5e651a08-c333-4397-a274-7882db52cc50.png)